apps/api/src/index.ts

```diff
@@ -40,7 +40,22 @@ const main = async (): Promise<void> => {
         metadataProvider,
         logger,
     );
-    const metricsController = new MetricsController(l1MetricsService, metadataProvider, logger);
+
+    const l2ChainsConfigMap = config.l2;
+    const l2MetricsMap = new Map<ChainId, L2MetricsService>();
+
+    for (const [chainId, rpcUrls] of Object.entries(l2ChainsConfigMap)) {
+        const provider = new ZKChainProvider(rpcUrls, logger);
+        const metricsService = new L2MetricsService(provider, logger);
+        l2MetricsMap.set(BigInt(chainId), metricsService);
+    }
+
```

Comment:
i was wondering if it would be better to encapsulate this logic inside the L2MetricsService and you just pass the map as parameter. But this approach seems to be more decoupled. 👍